### DISCUSSION TITLE
Behavior of Java Agent during OutOfMemory errors?

### DISCUSSION THREAD

**Discussion by dajulia3** [open-telemetry/opentelemetry-java-instrumentation#2414] [Q&A]
Hi all, 

I'm wondering what happens in the case of an OutOfMemoryError in application code. Does the agent add and error to the current span and transmit it, or is it generally unable to do so? I'd appreciate it if someone could provide some insight here :) 

Thanks!

**Comment by trask** [comment]
hey @dajulia3, if the `OutOfMemoryError` bubbles up to the operation that is being traced (e.g. servlet span), then the javaagent will attach it to the span.

with the generally caveat that all bets are off for everything in the JVM when `OutOfMemoryError` occurs. (btw highly recommend `-XX:+HeapDumpOnOutOfMemoryError` and Eclipse MemoryAnalyzer for these)
